### ISSUE TITLE
Save html for custom checkbox when setting custom for: and value:

### ISSUE THREAD

**Issue by dachinat** [bootstrap-ruby/bootstrap_form#471]
I needed to change an id of checkbox input, so I had to change 'for' of the label as well.

`<%= f.check_box :cstm_ns, { custom: true, id: "custom_id", label: { value: "&#8203;".html_safe, for: "custom_id}" } }, smth.id, 0 %>`

When using `label: { value: .. }` html was escaped, so I monkey patched `check_box` method in `BootstrapForm::FormBuilder`
```

if options[:custom]
  div_class = ["custom-control", "custom-checkbox"]
  div_class.append("custom-control-inline") if layout_inline?(options[:inline])
  label_class = label_classes.prepend("custom-control-label").compact.join(" ")
  content_tag(:div, class: div_class.compact.join(" ")) do
    html = if options[:skip_label]
             checkbox_html
           else
             # TODO: Notice we don't seem to pass the ID into the custom control.

             #
             # This is what was changed
             #
             form_label = label_description.is_a?(Hash) ?
                              content_tag(:label,
                                          label_description[:value],
                                          label_description.except(:value).merge(
                                              class: "custom-control-label"
                                          ),
                                          false, &block) :
                              label(label_name, label_description, class: label_class)

             checkbox_html.concat(form_label)
             #
             #
             #
           end
    html.concat(generate_error(name)) if options[:error_message]
    html
  end
else
```

**Comment by lcreid** [comment]
Thanks for finding this! I can't think of any reason why this shouldn't work for custom controls. I'll put together a fix for this shortly.

**Comment by mattbrictson** [comment]
Fixed via #472 and 3032ba0
